students/perrymant/blog-01-hello-world-deconstructed.md

```diff
@@ -0,0 +1,347 @@
+Part of the reason for this level of verbosity is due to the fact that Java is forcing you to adopt the `Object Oriented Programming` paradigm, in which a program seeks to represent the real-world as a set of `objects`.
+
+## Term 2: `class`
+
+This is a good point to discuss the second term `class`.
+`class` is a reserved keyword in Java, and like all other reserved keywords, it must start with a lower case letter, but more importantly: it has a very specific purpose.
+A `class` serves as a blueprint for creating `objects`, and it will define their features (`attributes`) and what they can do (`behaviour`). To make a `class`, all we have to do is write the keyword `class` followed by the name of the class, and then we can add the attributes and behaviour inside the code-block which is contained within the curly brackets.
+Here are two examples:
+
+```java
+class Frog {
+
+}
+```
+
+```java
+class Fly {
+
+}
+```
+We've just made a `Frog class` and a `Fly class`!
+Notice the initial capital letter - another Java naming convention that we must adhere to for class names.
+Also notice that the words used for classes should be nouns, as they describe either a person, entity, thing, or idea.
+You can work out from this that in our main example, we have a `class` called `HelloWorld`, indicated by the third term.
+In fact, not only is the class called `HelloWorld`, the file that this bit of code lives in **HAS** to be called `HelloWorld.java`.
+We'll see in a moment exactly why this has to be the case for our example, but typically, each file will just have one class in it, and the name of the file and the class must be the same (of course, the file will also include the .java extension).
+So with our `Frog` and `Fly` classes, the file and class structure will look like this:
+```java
++---Frog.java--------+     +---Fly.java------------------+
+|                    |     |                             |
+|  class Frog {      |     |  class Fly {                |
+|      int eyes = 2; |     |      void irritateHuman() {  |
+|                    |     |                             |
+|  }                 |     |      }                      |
+|                    |     |  }                          |
+|                    |     |                             |
++--------------------+     +-----------------------------+
+```
+
+I've taken the liberty to give the `Frog` the attribute of two `eyes`, and I've given the `Fly` the `irritateHuman()` behaviour.
+So you see how each class is contained in its own file, and the class and filename are the same.
+This is useful because it keeps everything that is specific to the `Frog` in one place, and everything that is specific to the `Fly` in it's own place, which can help us out in the long run by making our overall program more navigatable and readable.
+This concept is refered to as `encapsulation`, allowing these related ideas to be put together into one unit.
```

Comment:
I quite like this wording to explain encapsulation. I think it's perhaps a more important point than the more common explanation of encapsulation, although I do think it's worth mentioning that here too. The way I generally hear it being described is "preventing unrelated things from having access".